src/app/integrations/edit-page/step-configure/step-configure.component.ts

```diff
@@ -126,6 +128,24 @@ export class IntegrationsStepConfigureComponent extends FlowPage
     });
   }
 
+  fetchDataShapesFor(step: Step) {
+    return this.integrationSupport.fetchMetadata(
+      step.connection,
+      step.action,
+      step.configuredProperties || {},
+    )
+    .toPromise()
+    .then(response => {
+      log.debug('Response: ' + JSON.stringify(response, undefined, 2));
+      const definition: any = response['_body'] ? JSON.parse(response['_body']) : undefined;
+      step.action.inputDataShape = definition.inputDataShape;
+      step.action.outputDataShape = definition.outputDataShape;
+    })
+    .catch(response => {
```

Comment:
Here we could receive a user-friendly error message and use the NotificationService to show it.